cli/src/main/kotlin/tools/samt/cli/DiagnosticFormatter.kt

```diff
@@ -0,0 +1,505 @@
+        }
+
+        private fun rowContextLowerBound(row: Int): Int = maxOf(0, row - CONTEXT_ROW_COUNT)
+        private fun rowContextUpperBound(row: Int, sourceFile: SourceFile): Int = minOf(sourceFile.sourceLines.lastIndex, row + CONTEXT_ROW_COUNT)
+    }
+
+    private fun format(): String = buildString {
+
+        // print context-less messages
+        val globalMessages = diagnosticController.globalMessages
+        globalMessages.forEachIndexed { index, message ->
+            appendLine("⎯".repeat(terminalWidth))
+            append(formatGlobalMessage(message))
+            if (index != globalMessages.lastIndex) {
+                appendLine()
+            }
+        }
+
+        // print file-specific messages
+        for (context in diagnosticController.contexts) {
+
+            // sort messages by severity and then by location (row)
+            context.messages.sortWith(compareBy(
+                { it.severity },
+                { it.highlights.firstOrNull()?.location?.start?.row }
+            ))
+
+            context.messages.forEachIndexed { index, message ->
+                appendLine("⎯".repeat(terminalWidth))
+                append(formatMessage(message, context))
+                if (index != context.messages.lastIndex) {
+                    appendLine()
+                }
+            }
+        }
+    }
+
+    private fun severityForegroundColor(severity: DiagnosticSeverity): TextColors = when (severity) {
+        DiagnosticSeverity.Error -> white
+        DiagnosticSeverity.Warning -> black
+        DiagnosticSeverity.Info -> white
+    }
+
+    private fun severityBackgroundColor(severity: DiagnosticSeverity): TextColors = when (severity) {
+        DiagnosticSeverity.Error -> red
+        DiagnosticSeverity.Warning -> yellow
+        DiagnosticSeverity.Info -> blue
+    }
+
+    private fun formatTextForSeverity(text: String, severity: DiagnosticSeverity, withBackground: Boolean = false, withBold: Boolean = false): String {
+        val colorFormatted = if (withBackground) {
+            (severityForegroundColor(severity) on severityBackgroundColor(severity))(text)
+        } else {
+            (severityBackgroundColor(severity))(text)
+        }
+
+        return if (withBold) {
+            bold(colorFormatted)
+        } else {
+            colorFormatted
+        }
+    }
+
+    private fun formatSeverityIndicator(severity: DiagnosticSeverity): String = when (severity) {
+        DiagnosticSeverity.Error -> formatTextForSeverity("ERROR:", severity, withBold = true)
+        DiagnosticSeverity.Warning -> formatTextForSeverity("WARNING:", severity, withBold = true)
+        DiagnosticSeverity.Info -> formatTextForSeverity("INFO:", severity, withBold = true)
+    }
+
+    private fun formatFilePathRelativeToWorkingDirectory(filePath: String): String {
+        val workingDirectory = diagnosticController.workingDirectoryAbsolutePath
+        return filePath.removePrefix(workingDirectory).removePrefix("/")
```

Comment:
Use OS-agnostic constant
```suggestion
        return filePath.removePrefix(workingDirectory).removePrefix(File.separator)
```